src/sagas/import-project.saga.js

```diff
@@ -0,0 +1,160 @@
+import { call, put, cancel, select, takeEvery } from 'redux-saga/effects';
+import {
+  importExistingProjectStart,
+  importExistingProjectFinish,
+  importExistingProjectError,
+  SHOW_IMPORT_EXISTING_PROJECT_PROMPT,
+  IMPORT_EXISTING_PROJECT_START,
+} from '../actions';
+import {
+  loadPackageJson,
+  writePackageJson,
+} from '../services/read-from-disk.service';
+import { getColorForProject } from '../services/create-project.service';
+import { getInternalProjectById } from '../reducers/projects.reducer';
+
+const { dialog } = window.require('electron').remote;
+
+/**
+ * Handle path to project
+ * @param {Array<string>} paths
+ */
+export function* handlePathInput(paths) {
+  // The user might cancel out without selecting a directory.
+  // In that case, do nothing.
+  if (!paths) yield cancel();
+
+  // Only a single path should be selected
+  const [path] = paths;
+  yield put(importExistingProjectStart(path));
+}
+
+/**
+ * Show import dialog
+ */
+export function* showImportDialog() {
+  const paths = yield call([dialog, 'showOpenDialog'], {
+    message: 'Select the directory of an existing React app',
+    properties: ['openDirectory'],
+  });
+  yield call(handlePathInput, paths);
+}
+
+/**
+ * Show alert with error message that depends from error type
+ * @param {Error} err
+ */
+export function* handleImportError(err) {
+  switch (err.message) {
+    case 'project-name-already-exists': {
+      yield call(
+        [dialog, 'showErrorBox'],
+        'Project name already exists',
+        "Egad! A project with that name already exists. Are you sure it hasn't already been imported?"
+      );
+      break;
+    }
+
+    case 'unsupported-project-type': {
+      yield call(
+        [dialog, 'showErrorBox'],
+        'Unsupported project type',
+        "Looks like the project you're trying to import isn't supported. Unfortunately, Guppy only supports projects created with create-react-app or Gatsby"
+      );
+      break;
+    }
+
+    default: {
+      yield call([console, 'error'], err);
+      yield call(
+        [dialog, 'showErrorBox'],
+        'Unknown error',
+        'An unknown error has occurred. Sorry about that! Details have been printed to the console.'
+      );
+      break;
+    }
+  }
+}
+
+export function* importProject({ path }) {
+  try {
+    // Let's load the basic project info for the path specified, if possible.
+    const json = yield call(loadPackageJson, path);
+    const projectId = json.name;
+
+    // Check to see if we already have a project with this ID.
+    // In the future, maybe I can attach a suffix like `-copy`, but for
+    // now I'll just reject it outright.
+    const isAlredyExist = yield select(getInternalProjectById, projectId);
+    if (isAlredyExist) throw new Error('project-name-already-exists');
```

Comment:
Nit: for consistency with the rest of the codebase, can we avoid 1-line conditionals?

```js
if (isAlreadyExist) {
  throw new Error('project-name-already-exists');
}
```

Reply:
Sure, looks like a rule for eslint ;)